### PR TITLE
Bmb/gold

### PR DESCRIPTION
**Pull request by bbartman**
Implemented parsing for doc attributes and for tags/markup as primary expressions.

### CODE REVIEW

File: clang/lib/Gold/GoldDependentExprTransformer.cpp [hunk 1]
```diff
@@ -118,12 +118,15 @@ clang::Expr *DependentExprTransformer::transformCppxDependentMemberAccessExpr(
 
   clang::Expr *Ret = transformDependentExpr(E->getBase());
   if (!Ret) {
+    llvm::outs() << "We have an invalid dependent expression that was evaluated!\n";
```
Comment (sdgoodrick) [inline clang/lib/Gold/GoldDependentExprTransformer.cpp]: use a custom diagnostic, they're all over the ExprElaborator. and please use language that actually sounds like a diagnostic.

File: clang/lib/Gold/GoldLexer.cpp [hunk 2]
```diff
@@ -341,7 +369,9 @@ Token CharacterScanner::matchNewline() {
 }
 
 Token CharacterScanner::matchLineComment() {
-  assert(nextCharacterIs('#'));
+  // assert(nextCharacterIs('#'));
```
Comment (sdgoodrick) [inline clang/lib/Gold/GoldLexer.cpp]: can you clean up some of the comments in here? it's already a big change to look through

File: clang/lib/Gold/GoldParser.cpp [hunk 3]
```diff
@@ -57,10 +65,13 @@ TokenKind CloseTokens[]
...
   tok::Dedent,
   tok::Greater,
+  tok::HashGreater,
+  tok::Greater,
+  tok::Greater,
 };
 
 /// A class to help match enclosing tokens.
-template<EnclosureKind K>
+template<EnclosureKind K, bool NoFetchExpect = false>
```
Comment (sdgoodrick) [inline clang/lib/Gold/GoldParser.cpp]: add to the comment what NoFetchExpect does. assuming it stops from fetching a token? in that case, ExpectNoFetch is a much clearer name

File: clang/lib/Gold/GoldParser.cpp [hunk 4]
```diff
@@ -262,7 +555,9 @@ void Parser::parseArray(ArraySemantic S, llvm::SmallVectorImpl<Syntax *> &Vec) {
   appendTerm(Vec, List);
 
   while (true) {
-    // Obviously, stop at the end of the file.
```
Comment (sdgoodrick) [inline clang/lib/Gold/GoldParser.cpp]: why add a bunch of whitespace here?

File: clang/lib/Gold/GoldLexer.cpp (hunk 2 above)
Comment (bbartman) [reply clang/lib/Gold/GoldLexer.cpp]: Fixed, but not pushed yet.

File: clang/lib/Gold/GoldDependentExprTransformer.cpp (hunk 1 above)
Comment (bbartman) [reply clang/lib/Gold/GoldDependentExprTransformer.cpp]: Fixed but not pushed.

File: clang/lib/Gold/GoldParser.cpp (hunk 4 above)
Comment (bbartman) [reply clang/lib/Gold/GoldParser.cpp]: I was doing something and removed it then forgot to fix thing. I have this fixed but not pushed yet.

**Comment by bbartman** [comment]
This is ready to be merged.

**Pull request merged**